web-common/src/features/dashboards/time-dimension-details/charts/TDDAlternateChart.svelte

```diff
@@ -45,6 +51,20 @@
   $: comparedDimensionLabel =
     $comparisonDimension?.label || $comparisonDimension?.name;
 
+  $: hoveredTime = $tableInteractionStore.time;
+  $: hoveredDimensionValue = $tableInteractionStore.dimensionValue;
+
+  $: {
+    updateVegaOnTableHover(
+      viewVL,
+      chartType,
+      isTimeComparison,
+      hasDimensionData,
+      hoveredTime,
+      hoveredDimensionValue,
+    );
```

Comment:
**Optimization Issue**: The reactive block for 'updateVegaOnTableHover' is triggered excessively without checks on the actual changes to its dependencies, potentially leading to performance degradation due to unnecessary computations. <br> **Fix**: Add a conditional check to ensure that 'updateVegaOnTableHover' is only called when relevant dependencies change. <br> **Code Suggestion**: 
 ```
 -  $: {
 -    updateVegaOnTableHover(viewVL, chartType, isTimeComparison, hasDimensionData, hoveredTime, hoveredDimensionValue);
 -  }
 +  $: if ($tableInteractionStore.time !== previousTime || $tableInteractionStore.dimensionValue !== previousDimensionValue) {
 +    updateVegaOnTableHover(viewVL, chartType, isTimeComparison, hasDimensionData, hoveredTime, hoveredDimensionValue);
 +    previousTime = $tableInteractionStore.time;
 +    previousDimensionValue = $tableInteractionStore.dimensionValue;
 +  }
 ```